app/models/service_retire_task.rb

```diff
@@ -25,41 +25,43 @@ def after_request_task_create
     update_attributes(:description => get_description)
     parent_svc = Service.find_by(:id => options[:src_ids])
     _log.info("- creating service tasks for service <#{self.class.name}:#{id}>")
-
-    create_retire_subtasks(parent_svc)
+    create_retire_subtasks(parent_svc, self)
   end
 
-  def create_retire_subtasks(parent_service)
-    parent_service.direct_service_children.each { |child| create_retire_subtasks(child) }
+  def create_retire_subtasks(parent_service, parent_task)
     parent_service.service_resources.collect do |svc_rsc|
       next unless svc_rsc.resource.try(:retireable?)
       # TODO: the next line deals with the filtering for provisioning
       # (https://github.com/ManageIQ/manageiq/blob/3921e87915b5a69937b9d4a70bb24ab71b97c165/app/models/service_template/filter.rb#L5)
       # which should be extended to retirement as part of later work
       # svc_rsc.resource_type != "ServiceTemplate" || self.class.include_service_template?(self, srr.id, parent_service)
-      nh = attributes.except("id", "created_on", "updated_on", "type", "state", "status", "message")
+
+      # TODO: use changes here: https://github.com/ManageIQ/manageiq/pull/17996 to not have to filter by col_names
+      # 17996 removes virtual attributes from @attrs list
+      nh = attributes.slice(self.class.column_names).except!("id", "created_on", "updated_on", "type", "state", "status", "message")
```

Comment:
@d-m-u `slice` expects `*keys` not an array of names.  (https://apidock.com/rails/Hash/slice)

You need `nh = attributes.slice(*self.class.column_names)...`

When I run the command the way it is here I get an empty hash.  Now my concern is that tests are passing with invalid data.  😱